app/frontend/src/features/messages/utils.ts

```diff
@@ -0,0 +1,43 @@
+import { Message } from "../../pb/conversations_pb";
+import { hostRequestStatusLabels } from "./constants";
+
+export function isControlMessage(message: Message.AsObject) {
+  return !message.text;
+}
+
+export function messageTargetId(message: Message.AsObject) {
+  return message.userInvited
+    ? message.userInvited.targetUserId
+    : message.userMadeAdmin
+    ? message.userMadeAdmin.targetUserId
+    : message.userRemovedAdmin
+    ? message.userRemovedAdmin.targetUserId
+    : undefined;
+}
+
+export function controlMessageText(
+  message: Message.AsObject,
+  authorName: string,
+  targetName?: string
+) {
+  const authorCap = authorName.charAt(0).toUpperCase() + authorName.slice(1);
+  if (message.chatCreated !== undefined) {
```

Comment:
Couldn't we shorten these checks to `if (message.chatCreated)` etc, since it should be fine to just check for its truthiness?